packages/flame/lib/src/game/game_render_box.dart

```diff
@@ -2,14 +2,55 @@ import 'package:flame/src/game/game_loop.dart';
 import 'package:flame/src/game/mixins/game.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart' hide WidgetBuilder;
-//ignore_for_file: unnecessary_non_null_assertion
+
+class RenderGameWidget extends LeafRenderObjectWidget {
+  final Game game;
+
+  const RenderGameWidget({
+    Key? key,
+    required this.game,
+  }) : super(key: key);
+
+  @override
+  RenderBox createRenderObject(BuildContext context) {
+    return GameRenderBox(game, context);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, GameRenderBox renderObject) {
+    renderObject
+      ..game = game
+      ..buildContext = context;
+  }
+}
 
 class GameRenderBox extends RenderBox with WidgetsBindingObserver {
-  BuildContext buildContext;
-  Game game;
+  GameRenderBox(this._game, this.buildContext);
+
   GameLoop? gameLoop;
 
-  GameRenderBox(this.buildContext, this.game);
+  BuildContext buildContext;
+
+  Game _game;
+
+  Game get game => _game;
+
+  set game(Game value) {
+    // Identities are equal, no need to update.
```

Comment:
I think this comment is not needed.

I usually tend to dislike code comments because code gets updated but comments are not guaranteed to do so. Usually comments are a hint that the code is not verbose enough.


Anyways, I think it is pretty clear that the following snippet doesn't update the game if the value is the same as the current game.
```dart
if (_game == value) {
      return;
    }
```